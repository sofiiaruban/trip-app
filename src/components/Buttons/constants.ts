export enum ButtonColor {
  PRIMARY = 'primary',
  BASIC = 'basic'
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset'
}
