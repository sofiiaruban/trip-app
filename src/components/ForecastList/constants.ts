export enum ForecastCardType {
  SMALL = 'small',
  BIG = 'big'
}
