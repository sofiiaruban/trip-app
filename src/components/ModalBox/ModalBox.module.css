.modalOverlay {
  position: fixed;
  height: 100%;
  width: 100vw;
  top: 0px;
  left: 0;
  background-color: rgba(24, 24, 24, 0.8);
  z-index: 4;
  display: flex;
  justify-content: center;
}
.modalBox {
  width: 500px;
  height: fit-content;
  border-radius: 4px;
  z-index: 5;
  position: relative;
  top: 80px;
  background-color: #fff;
}
.modalBoxHead {
  padding: 20px 0 10px 24px;
  border-bottom: 1px solid rgba(233, 229, 229, 0.8);
  margin-bottom: 1rem;
}
.modalBoxTitle {
  margin-top: 0;
  font-weight: 600;
  text-align: start;
  padding: 0;
}
.buttonClose {
  box-sizing: border-box;
  border: none;
  position: absolute;
  top: 18px;
  right: 16px;
}
