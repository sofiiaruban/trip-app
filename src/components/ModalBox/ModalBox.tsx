import { FC, MouseEventHandler, ReactNode } from 'react'
import { createPortal } from 'react-dom'
import { IoClose } from 'react-icons/io5'
import styles from './ModalBox.module.css'
import ButtonIcon from '../Buttons/ButtonIcon'

interface ModalProps {
  title: string
  isOpen: boolean
  children: ReactNode
  closeModal: MouseEventHandler<HTMLElement> 
}

const ModalBox: FC<ModalProps> = ({ isOpen, title, children, closeModal }) => {
  if (!isOpen) return null

  const modalBoxPropagationHandle: MouseEventHandler<HTMLDivElement> = (
    event
  ) => {
    event.stopPropagation()
  }

  return createPortal(
    <div onClick={closeModal} className={styles.modalOverlay}>
      <div
        onClick={modalBoxPropagationHandle}
        className={styles.modalBox}
        role="dialog"
      >
        <div className={styles.modalBoxHead}>
          <p className={styles.modalBoxTitle}>{title}</p>
          <div className={styles.buttonClose}>
            <ButtonIcon onClick={closeModal}>
              <IoClose />
            </ButtonIcon>
          </div>
        </div>
        {children}
      </div>
    </div>,
    document.body
  )
}

export default ModalBox
