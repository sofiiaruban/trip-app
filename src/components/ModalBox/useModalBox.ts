import { disableBodyScroll, enableBodyScroll } from 'body-scroll-lock'
import { useEffect, useState } from 'react'

interface ModalHookReturnType {
  openModal: () => void
  closeModal: () => void
  isOpen: boolean
}

const useModal = (): ModalHookReturnType => {
  const [isOpen, setIsModalOpen] = useState(false)

  useEffect(() => {
    if (isOpen) {
      disableBodyScroll(document.body)

      return () => {
        enableBodyScroll(document.body)
      }
    }
  }, [isOpen])

  const openModal = () => {
    setIsModalOpen(true)
  }

  const closeModal = () => {
    setIsModalOpen(false)
  }

  return {
    openModal,
    closeModal,
    isOpen
  }
}

export default useModal
