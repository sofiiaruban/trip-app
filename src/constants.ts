export enum CityList {
  NEW_YORK = 'New York',
  LONDON = 'London',
  KYIV = 'Kyiv'
}

export enum CityImg {
  NEW_YORK = 'src/assets/newyork.jpeg',
  LONDON = 'src/assets/london.jpg',
  KYIV = 'src/assets/kyiv.jpg'
}
