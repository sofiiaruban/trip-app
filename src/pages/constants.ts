export const MAIN_PAGE_TEST_ID = 'mainPage'
