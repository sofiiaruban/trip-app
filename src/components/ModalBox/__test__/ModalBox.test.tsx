import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import ModalBox from '../ModalBox'
import { MODAL_BOX_TEST_ID } from '../constants'

const commonProps = {
  isOpen: true,
  title: 'Test Modal'
}
const closeModalMock = jest.fn()

const testChild = <div data-testid="test-child">Hello, World!</div>

test('ModalBox component matches snapshot when open', async () => {
  const { getByTestId } = render(
    <ModalBox closeModal={closeModalMock} {...commonProps}>
      {testChild}
    </ModalBox>
  )
  const modalOverlay = getByTestId(MODAL_BOX_TEST_ID)
  expect(modalOverlay).toBeInTheDocument()
  expect(document.body).toMatchSnapshot()
})
