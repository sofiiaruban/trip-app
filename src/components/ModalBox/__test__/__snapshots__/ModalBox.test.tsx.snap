// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBox component matches snapshot when open 1`] = `
<body>
  <div />
  <div
    class="modalOverlay"
    data-testid="modalBoxComponent"
  >
    <div
      class="modalBox"
      role="dialog"
    >
      <div
        class="modalBoxHead"
      >
        <p
          class="modalBoxTitle"
        >
          Test Modal
        </p>
        <div
          class="buttonClose"
        >
          <button
            class="buttonIcon"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m289.94 256 95-95A24 24 0 0 0 351 127l-95 95-95-95a24 24 0 0 0-34 34l95 95-95 95a24 24 0 1 0 34 34l95-95 95 95a24 24 0 0 0 34-34z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        data-testid="test-child"
      >
        Hello, World!
      </div>
    </div>
  </div>
</body>
`;
